SparkPost/2web2ui
# SA-1587: Delete a saved report

### What Changed
 - Added delete action/reducer
 - Added delete modal
 - Added tests
 - Fixed some older bugs
### How To Test
- Go to report builder in Hibana. Enable saved reports with UI user option `allow_saved_reports`.
- Make a new saved report through the save new report button.
- Once that succeeds, click the view all reports button.
- Find your report in the table. Click on the actionlist (...) and then click delete.
- Confirm the deletion by clicking Delete in the new modal.
